eclass/user-info.eclass

```diff
@@ -145,7 +158,16 @@ egetgroups() {
 	[[ $# -eq 1 ]] || die "usage: egetgroups <user>"
 
 	local egroups_arr
-	read -r -a egroups_arr < <(id -G -n "$1")
+	if [[ -n "${ROOT}" ]]; then
+		local pgroup=$(grep "^$1:" ${ROOT}/etc/passwd | cut -d: -f1)
```

Comment:
you should use `pgroup="$(egetent passwd $1)"`

Reply:
also i don't think this works, the -f is wrong here, it should be -f4, which gets the gid, then use egetgroupname $gid

Reply:
-f1 get the group name, -f4 will get the gid.
Why can't I use -f1 then ? Is it different for BSD systems ?